src/figuresClasses.ts

```diff
@@ -1,19 +1,118 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
+}
 
+export function roundArea(area: number): number {
+  return Math.floor((area) * 100) / 100;
 }
 
-export class Triangle {
+abstract class Basic implements Figure {
+  abstract readonly shape: Shape;
+
+  abstract color: Color;
 
+  abstract getArea(): number;
+
+  checkAllSides(sides: number[]): void {
+    if (sides.some((side) => side <= 0)) {
+      throw new Error(
+        `Figure of shape ${this.shape} should have only positive params`,
+      );
+    }
+  }
 }
 
-export class Circle {
+export class Triangle extends Basic {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    this.checkAllSides([a, b, c]);
```

Comment:
Or you can use rest operator in this function to be able pass any amount of params (up to you)